packages/dialtone-vue2/components/emoji_picker/emoji_picker.vue

```diff
@@ -10,9 +10,8 @@
         :scroll-into-tab="scrollIntoTab"
         :tab-set-labels="tabSetLabels"
         :is-scrolling="isScrolling"
-        @tab-key-pressed="focusNextSectionFromEmojiTabSet"
-        @shift-tab-key-pressed="$refs.skinSelectorRef.focusSkinSelector()"
-        @arrow-down-key-pressed="focusNextSectionFromEmojiTabSet"
+        @focus-skin-selector="$refs.skinSelectorRef.focusSkinSelector()"
+        @focus-search-input="showSearch ? $refs.searchInputRef.focusSearchInput() : $refs.emojiSelectorRef.focusEmojiSelector()"
```

Comment:
I don't think it makes it clearer to have a ternary in the template really, while a named function could be better. Maybe there's a better name than "focusNextSectionFromEmojiTabSet".
Also emitting `focus-search-input` is weird if we are not really going to focus the search input. I think the emoji selector shouldn't know if there is a search input or not. It's responsibility is to emit that the tab key was pressed. That is not how this was handled before that's why is inconsistent, but I think that approach is prone to errors (like the bug we just solved)

Reply:
Hiding search input is an optional case but the structure still has it. 
You are just... hiding it.

It is okay if you go with a structure event emits or an abstract event handler. But we should **not mix them**.

At this point the component responds to a structured keyboard navigation in order by

1. Tab
2. SearchInput
3. EmojiSelector
4. SkinSelector

Keyboard navigation is being handled by the components itself in order of this structure.
Thats why you can move around the component **and** jump to the previous or next section.

Hiding a part of this structure its something that the emoji-picker component is not prepared for.
We should extract this `**and** jump to the previous or next section.` logic to the parent component and create a proper structure to handle it.

I don't see the previous PR code take part of this, then I propose a simple patch for it.
Then later if this is needed and this becomes a common practice (hidding parts of emoji-picker) we can take care of it accordingly.

> I don't think it makes it clearer to have a ternary in the template really, while a named function could be better
I agree, we can create a method here, I tested something like have a `focusSearchInput` method that handles if it comes from `emoji-selector` or `emoji-tabset` it should go to one or another if the `showSearch` is false.
But I see the ternary more efficient. 

> Also emitting focus-search-input is weird if we are not really going to focus the search input.
It's what the sections are supposed to do, thats why we have `@focus-` events. 
What is weird for the component is to hide a section.

Reply:
This external search definitely throws an annoying wrench into things.

It might make sense to emit keyboard events in more generic cases, however the action performed here has logic behind it specifically meaning "focus the input". If we were to emit and capture this event as a consumer it should still be used to "focus the input".

My question is, should we be setting focus at all when showSearch is false? Shouldn't we just emit the event and allow the consumer to focus the search input since it is external? 

I'm not sure we need to spend a ton of time to do this perfectly. We just need the ability to fix this one weird case where we have to use an external input. I doubt this event would be used in any case other than this one.

Reply:
We can expose some of this event later if it is required, for now they are all internal just for the correct behaviour of the component.

I will move forward this change, if there is some issue that this fix didn't take we have https://github.com/dialpad/dialtone/pull/389 implementation.